### PR TITLE
feat: add `preact` & `preact-ts` templates back

### PR DESCRIPTION
**Pull request by rschristian**
### What kind of change does this PR introduce?

Was given the go-ahead here: https://github.com/tauri-apps/create-tauri-app/issues/365#issuecomment-1626408630

- [ ] Bugfix
- [X] Feature
- [ ] Docs
- [ ] New Binding issue #___
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

### Does this PR introduce a breaking change?

- [ ] Yes, and the changes were approved in issue #___
- [X] No

### Checklist
- [ ] When resolving issues, they are referenced in the PR's title (e.g `fix: remove a typo, closes #___, #___`)
- [X] A change file is added if any packages will require a version bump due to this PR per [the instructions in the readme](https://github.com/tauri-apps/create-tauri-app/blob/dev/.changes/readme.md).
- [X] I have added a convincing reason for adding this feature, if necessary

### Other information

Seems to work well, but this is my first time contributing to `create-tauri-app` (actually, my first time using `tauri` at all, despite it being on my to-do list since forever), so certainly let me know if I missed something or did something silly!

As stated in the issue, I'm happy to help out & maintain these going forward, if any changes are needed or if Preact-specific knowledge is required. I don't want to increase your maintenance burden.

### CODE REVIEW

File: packages/cli/src/template.rs [hunk 1]
```diff
@@ -47,6 +49,7 @@ impl Template {
             Template::Svelte => "Svelte - (https://svelte.dev/)",
             Template::React => "React - (https://reactjs.org/)",
             Template::Solid => "Solid - (https://www.solidjs.com/)",
+            Template::Preact => "Preact - (https://preactjs.com/)",
```
Comment (rschristian) [inline packages/cli/src/template.rs]: The order here (especially in regards to Angular) was something I was unsure about: generally I've appended Preact after Solid, as it made sense to group alongside the other `framework` & `framework-ts` style options, but appending it after Solid here means it goes above Angular in the list of framework options a user will see.

Is it better to keep the consistency (after Solid) or keep order of addition, and move it to the bottom (below Angular)? Happy to do whatever, just let me know!

Comment (amrbashir) [reply packages/cli/src/template.rs]: It should be added at the end, new templates always go last

Comment (rschristian) [reply packages/cli/src/template.rs]: Does that go for everything, including places like [this](https://github.com/rschristian/create-tauri-app/blob/e7a9992db2de0147e2773e5fc3a4d361ad94ed7d/packages/cli/src/template.rs#L112-L127)?

Obviously not user-facing, but I don't know if you'd prefer that to still be after Angular or if grouping with the similar templates is correct.

Comment (amrbashir) [reply packages/cli/src/template.rs]: yeah pretty much everywhere

Comment (rschristian) [reply packages/cli/src/template.rs]: Will do then, thanks!

Comment (rschristian) [reply packages/cli/src/template.rs]: Should be good now.

**Review by amrbashir — APPROVED** [review]
Thank you

**Pull request merged**